### ISSUE TITLE
[BUG] - Panic for Interim character

### ISSUE THREAD

**Issue by igorkova** [personnummer/go#31]
**Context**
Luhn is panicking when there is specific character within personnummer.
I'd say it's not handling interim characters properly but hard for me to know exactly what's going on.

**Expected outcome**

I am expecting that `personnummer.Valid("12120111X3")` returns false

**Actual outcome**

it panics at https://github.com/personnummer/go/blob/master/personnummer.go#L115

```

--- FAIL: TestCustom (0.00s)
panic: runtime error: index out of range [40] with length 10 [recovered]
	panic: runtime error: index out of range [40] with length 10

goroutine 34 [running]:
testing.tRunner.func1.2({0x10130d2c0, 0x14000262000})
	/opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/testing.go:1399 +0x378
panic({0x10130d2c0, 0x14000262000})
	/opt/homebrew/Cellar/go/1.19.1/libexec/src/runtime/panic.go:884 +0x204
github.com/personnummer/go/v3.luhn(...)
	...temp/personnummer/personnummer.go:115
github.com/personnummer/go/v3.(*Personnummer).valid(0x1400024e000)
```

**Test case**
```
func TestCustom(t *testing.T) {
	assert.Equal(t, false, Valid("12120111X3"))
}
```

**Comment by frozzare** [comment]
Thanks for the bug report, i'll take a look when I have time. Feel free to send a pull request :)

**Comment by frozzare** [comment]
Fixed in 3.1.1. 

The code for interim worked fine, but `12120111X3` is an invalid interim number since the `X` is in the wrong position and that's the reason luhn failed. It's now fixed.

A correct interim number looks like this `20000101T220` (long format) or `000101T220` (short format).
